### PR TITLE
[SPARK-34266][SQL][DOCS] Update comments for `SessionCatalog.refreshTable()` and `CatalogImpl.refreshTable()`

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
Describe `SessionCatalog.refreshTable()` and `CatalogImpl.refreshTable()`. what they do and when they are supposed to be used.

### Why are the changes needed?
To improve code maintenance.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
By running `./dev/scalastyle`

### CODE REVIEW

**Comment by MaxGekk** [comment]
@cloud-fan @sunchao @HyukjinKwon @dongjoon-hyun Please, review this PR.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala [hunk 1]
```diff
@@ -998,7 +998,26 @@ class SessionCatalog(
   }
 
   /**
-   * Refresh the cache entry for a metastore table, if any.
+   * Refresh table entries in structures maintained by the session catalog such as:
+   *   - The map of temporary or global temporary view names to their logical plans
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala:1002]: I think we don't refresh the map, we refresh the temp view logical plan in that map.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala [hunk 2]
```diff
@@ -998,7 +998,26 @@ class SessionCatalog(
...
+   *   - The relation cache which maps table/view identifiers to their logical plans
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala:1003]: does the relation cache apply to permanent views?

File: sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala [hunk 3]
```diff
@@ -515,14 +515,19 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
...
-   * Invalidates and refreshes all the cached data and metadata of the given table or view.
-   * For Hive metastore table, the metadata is refreshed. For data source tables, the schema will
-   * not be inferred and refreshed.
+   * The method fully refreshes a table or view with the given name including:
+   *   1. The relation cache in the session catalog. The method remove table entry from the cache.
+   *   2. The file indexes of all relations used by the given view.
+   *   3. Table/View schema in the Hive Metastore if the SQL config
+   *      `spark.sql.hive.caseSensitiveInferenceMode` is set to `INFER_AND_SAVE`.
+   *   4. Cached data of the given table or view, and all its dependents that refer to it.
+   *      Cached data is cleared while keeping the table/view and all its dependents are cached.
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala:524]: `keeping ... are cached` -> `keeping ... as cached`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala (hunk 2 above)
Comment (MaxGekk) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala:1003]: ah, no. Let me remove views.

**Comment by MaxGekk** [comment]
@sunchao @dongjoon-hyun WDYT about the updated comments.

File: sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala [hunk 4]
```diff
@@ -515,14 +515,19 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
   }
 
   /**
-   * Invalidates and refreshes all the cached data and metadata of the given table or view.
-   * For Hive metastore table, the metadata is refreshed. For data source tables, the schema will
-   * not be inferred and refreshed.
+   * The method fully refreshes a table or view with the given name including:
+   *   1. The relation cache in the session catalog. The method remove table entry from the cache.
```
Comment (sunchao) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala:519]: nit: remove -> removes

File: sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala [hunk 5]
```diff
@@ -515,14 +515,19 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
...
+   *   2. The file indexes of all relations used by the given view.
+   *   3. Table/View schema in the Hive Metastore if the SQL config
+   *      `spark.sql.hive.caseSensitiveInferenceMode` is set to `INFER_AND_SAVE`.
+   *   4. Cached data of the given table or view, and all its dependents that refer to it.
+   *      Cached data is cleared while keeping the table/view and all its dependents as cached.
```
Comment (sunchao) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala:524]: I'm a bit confused by this sentence, do you mean to say that existing cached data will be cleared and the cache will be lazily filled when the next time the table/view or the dependents are accessed?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala [hunk 6]
```diff
@@ -998,7 +998,26 @@ class SessionCatalog(
...
+   * For temp views, it refreshes their logical plans, and as a consequence of that it can refresh
+   * the file indexes of the base relations (`HadoopFsRelation` for instance) used in the views.
+   * The method still keeps the views in the internal lists of session catalog.
+   *
+   * For tables/views, it removes their entries from the relation cache.
+   *
+   * The method is supposed to use in the following situations:
+   *   1. The logical plan of a table/view was changed, and cached table/view data is cleared
+   *      explicitly. For example, like in `AlterTableRenameCommand` which re-caches the table
+   *      itself. Otherwise if you need to re-fresh cached data, consider using of
```
Comment (sunchao) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala:1014]: nit: re-fresh -> refresh?

**Review by sunchao — COMMENTED** [review]
Thanks @MaxGekk . Looking good just a few nits.

File: sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala (hunk 5 above)
Comment (MaxGekk) [reply sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala:524]: This is exactly what I mean ;-) Let me just replace the sentence by yours.

File: sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala [hunk 7]
```diff
@@ -515,14 +515,20 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
...
+   *      the next time the table/view or the dependents are accessed.
    *
-   * If this table is cached as an InMemoryRelation, re-cache the table and its dependents lazily.
+   * The method does not do:
+   *   - schema inference for file source tables
+   *   - statistics update
    *
-   * In addition, refreshing a table also clear all caches that have reference to the table
-   * in a cascading manner. This is to prevent incorrect result from the otherwise staled caches.
+   * The method is supposed to use in all cases when need to refresh table/view data and meta-data.
```
Comment (sunchao) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala:531]: nit: supposed to use -> supposed to be used

**Review by sunchao — APPROVED** [review]
LGTM (non-binding)!, with one nit.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**